### PR TITLE
test: Add unit tests for the UrlWalker and related components

### LINKED ISSUE #38844: [jest] URL Walker
- [ ] Test different number of URLs (1, 2, 5, 10?)
- [ ] assert hover over middle URL shows tooltip

### PR DESCRIPTION
**Pull request by ryan953**
Fixes #38844

### CODE REVIEW

File: static/app/components/replays/walker/splitCrumbs.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ function splitCrumbs({
   return crumbs.map((crumb, i) => (
     <SingleLinkSegment
       key={i}
-      path={firstUrl}
+      path={crumb.data?.to?.split('?')?.[0]}
```
Comment (ryan953) [inline static/app/components/replays/walker/splitCrumbs.tsx:63]: This was a bug

File: static/app/components/replays/walker/splitCrumbs.tsx [hunk 2]
```diff
@@ -82,11 +82,8 @@ function SingleLinkSegment({
     </Tooltip>
   );
   if (onClick) {
-    return (
-      <Link href="#" onClick={onClick}>
-        {content}
-      </Link>
-    );
+    // TODO(replays): Add a href that deeplinks to `crumb.timestamp`
+    return <Link onClick={onClick}>{content}</Link>;
```
Comment (ryan953) [inline static/app/components/replays/walker/splitCrumbs.tsx:86]: Having the `href="#"` in there meant that we could cmd+click and open a new window, pointing at the same url. It wasn't helpful.

We should be able to deeplink to the timestamp, which would be a good use of `href`. Next iteration

File: static/app/components/replays/walker/chevronDividedList.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    expect(await screen.queryByRole('listitem')).not.toBeInTheDocument();
+    expect(await screen.queryByRole('separator')).not.toBeInTheDocument();
+  });
+
+  it('should accept one item and show it in the <List>', async () => {
+    const mockItems = [<span key="1">first</span>];
+    render(<ChevronDividedList items={mockItems} />);
+
+    expect(await screen.queryByRole('listitem')).toBeInTheDocument();
```
Comment (scttcper) [inline static/app/components/replays/walker/chevronDividedList.spec.tsx:18]: queryBy doesn't return a promise

File: static/app/components/replays/walker/splitCrumbs.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    const results = splitCrumbs({
+      crumbs,
+      onClick,
+      startTimestampMs,
+    });
+    expect(results).toHaveLength(1);
+
+    render(results[0]);
+    expect(await screen.findByText('0 Pages')).toBeInTheDocument();
```
Comment (scttcper) [inline static/app/components/replays/walker/splitCrumbs.spec.tsx:45]: this component doesn't make any requests right? should be able to use getBy

**Review by scttcper — APPROVED** [review]

**Pull request merged**